matio: 1.5.27 fails to cross-compile (NixOS 24.11)

## Describe the bug

Attempting to cross build for Raspberry Pi 5 and `matio` fails to build. Apparently it is trying to run a check for `va_copy` with Autotools when cross building.

## Steps To Reproduce

Steps to reproduce the behavior:

1. `nix build .#pkgsCross.aarch64-multiplatform.matio`

## Expected behavior

Should build.

## Screenshots

```
error: builder for '/nix/store/n3rl7b0302lsr2qcv1bfrljqzyic14k6-matio-aarch64-unknown-linux-gnu-1.5.27.drv' failed with exit code 1;
       last 25 log lines:
       > checking size of char... 1
       > checking size of void *... 8
       > checking size of off_t... 8
       > checking size of size_t... 8
       > checking for mat_int64_t... int64_t
       > checking for mat_uint64_t... uint64_t
       > checking for mat_int32_t... int32_t
       > checking for mat_uint32_t... uint32_t
       > checking for mat_int16_t... int16_t
       > checking for mat_uint16_t... uint16_t
       > checking for mat_int8_t... int8_t
       > checking for mat_uint8_t... uint8_t
       > checking for off_t... yes
       > checking for off64_t... no
       > checking for long double... yes
       > checking for unsigned long long int... yes
       > checking for long long int... yes
       > checking for unsigned long long int... (cached) yes
       > checking for size_t... yes
       > checking for intmax_t... yes
       > checking for uintmax_t... yes
       > checking for uintptr_t... yes
       > checking for va_copy() function... configure: error: in `/build/matio-1.5.27':
       > configure: error: cannot run test program while cross compiling
       > See `config.log' for more details
       For full logs, run 'nix log /nix/store/n3rl7b0302lsr2qcv1bfrljqzyic14k6-matio-aarch64-unknown-linux-gnu-1.5.27.drv'.
error: 1 dependencies of derivation '/nix/store/99ahxb5sick3sy9ybyg0lwygk7p6z8zi-vips-aarch64-unknown-linux-gnu-8.15.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nf1vkhi2dmapx9bsj4yzynr686znv3pl-snapweb-aarch64-unknown-linux-gnu-0.8.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fd2pmcfznzhj3a90kcvxlv22x3s9fsd1-unit-snapserver.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6i1dyh4b3vhc9gmca9rghjy9s7fw4x9a-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fdsijrih44kjr9w1vykagli3x3sb2c70-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/h64h7b93pvvlfj9qfxdgq02lj2ilqj6r-nixos-system-piceiver-24.11.20241116.057f63b.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3fbs8kx83vaz7miyxvp0badjyf1aqmpr-ext4-fs.img.zst-aarch64-unknown-linux-gnu.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jkm7mkw3ir1y3v2bg4fx4d5y7szzi3kg-nixos-sd-image-24.11.20241116.057f63b-aarch64-linux.img-aarch64-unknown-linux-gnu.drv' failed to build
```

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.7-300.fc41.x86_64, Fedora Linux, 41.20241117.0 (Sway Atomic), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: aarch64-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/jordan/.config/nix/nix.conf:/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/lknx1arna0x37dg61ixsjbdlwcy9dvld-lix-2.91.1/share`
 - nixpkgs: `/nix/store/4r8s42c9mwfvgdlxv0izb1cmlzrsb5nz-source`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ma/matio/package.nix
+++ b/pkgs/by-name/ma/matio/package.nix
@@ -1,7 +1,14 @@
 {
   fetchurl,
+  hdf5,
   lib,
+  matio,
+  nix-update-script,
+  pkgconf,
   stdenv,
+  testers,
+  validatePkgConfig,
+  zlib,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -13,13 +20,39 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-naaYk0ohVprwWOY0hWRmb0UCnmwrCHjKDY+WCb93uNg=";
   };
 
+  configureFlags = [ "ac_cv_va_copy=1" ];
+
+  nativeBuildInputs = [
+    pkgconf
+    validatePkgConfig
+  ];
+
+  buildInputs = [
+    hdf5
+    zlib
+  ];
+
+  passthru = {
+    tests = {
+      pkg-config = testers.hasPkgConfigModules {
+        package = matio;
+        versionCheck = true;
+      };
+      version = testers.testVersion {
+        package = matio;
+      };
+    };
+    updateScript = nix-update-script { };
+  };
+
   meta = {
     changelog = "https://sourceforge.net/p/matio/news/";
     description = "C library for reading and writing Matlab MAT files";
     homepage = "http://matio.sourceforge.net/";
     license = lib.licenses.bsd2;
-    maintainers = [ ];
+    maintainers = with lib.maintainers; [ jwillikers ];
     mainProgram = "matdump";
     platforms = lib.platforms.all;
+    pkgConfigModules = [ "matio" ];
   };
 })

